bin/reth/src/args/multi_use_args.rs

```diff
@@ -0,0 +1,49 @@
+use crate::dirs::{PlatformPath, SecretKeyPath};
+use hex::encode as hex_encode;
+use reth_network::config::rng_secret_key;
+use secp256k1::{Error as SecretKeyBaseError, SecretKey};
+use std::fs::read_to_string;
+use thiserror::Error;
+
+/// Errors returned by loading a [`SecretKey`][secp256k1::SecretKey], including IO errors.
+#[derive(Error, Debug)]
+#[allow(missing_docs)]
+pub enum SecretKeyError {
+    #[error(transparent)]
+    SecretKeyDecodeError(#[from] SecretKeyBaseError),
+    #[error("An I/O error occurred: {0}")]
+    IOError(#[from] std::io::Error),
+}
+
+/// Attempts to load a [`SecretKey`] from a specified path. If no file exists
+/// there, then it generates a secret key and stores it in the default path. I/O
+/// errors might occur during write operations in the form of a
+/// [`SecretKeyError`]
+pub fn get_secret_key(
+    secret_key_path: &PlatformPath<SecretKeyPath>,
+) -> Result<SecretKey, SecretKeyError> {
+    let fpath = secret_key_path.as_ref();
+    let exists = fpath.try_exists();
+
+    match exists {
+        Ok(true) => {
+            let contents = read_to_string(fpath)?;
+            (contents.as_str().parse::<SecretKey>()).map_err(SecretKeyError::SecretKeyDecodeError)
+        }
+        Ok(false) => {
+            let default_path = PlatformPath::<SecretKeyPath>::default();
+            let fpath = default_path.as_ref();
+
+            if let Some(dir) = fpath.parent() {
+                // Create parent directory
+                std::fs::create_dir_all(dir)?
+            }
+
+            let secret = rng_secret_key();
+            let hex = hex_encode(secret.as_ref());
+            std::fs::write(fpath, hex)?;
+            Ok(secret)
```

Comment:
is this effectively a `cp` operation?

Reply:
So this implements what @gakonst recommended:

1. `Ok(true)`: If the secret key exists, then it reads the file.
2. `Ok(false)`: If the secret key does not exist but the path is accessible, then create a new secret key and write it to disk.
3. `Err(e)`: if the path cannot be reached (invalid permissions, etc), error.

I think it is a tad different than `cp`; it saves the key to disk if it doesn't exist and uses an existing key if the key does exist.

One consequence of this PR is that the default behavior changes. After the first `reth` call, the p2p secret will always be the same on subsequent runs and similarly the node peer ID will be static. On the main branch, currently the p2p secret and node ID changes every time `reth` is run.

Reply:
> After the first reth call, the p2p secret will always be the same on subsequent runs and similarly the node peer ID will be static.

hmm, i am a bit skeptical about this, imo this should always require opt-in.